src/main/java/com/commercetools/sync/products/utils/ProductVariantUpdateActionUtils.java

```diff
@@ -117,24 +129,81 @@
      * {@link ProductVariantDraft} and the {@link ProductVariant} have identical list of prices, then no update action
      * is needed and hence an empty {@link List} is returned.
      *
-     * <p>TODO: NOTE: Right now it always builds SetPrices UpdateAction, comparison should be
-     * TODO: calculated GITHUB ISSUE#101.
-     *
-     * <p>TODO: UNIT TESTS
-     *
      * @param oldProductVariant the {@link ProductVariant} which should be updated.
      * @param newProductVariant the {@link ProductVariantDraft} where we get the new list of prices.
+     * @param syncOptions the sync options wrapper which contains options related to the sync process supplied by
+     *                    the user. For example, custom callbacks to call in case of warnings or errors occurring
+     *                    on the build update action process. And other options (See {@link ProductSyncOptions}
+     *                    for more info.
      * @return a list that contains all the update actions needed, otherwise an empty list if no update actions are
      *         needed.
      */
     @Nonnull
     public static List<UpdateAction<Product>> buildProductVariantPricesUpdateActions(
         @Nonnull final ProductVariant oldProductVariant,
-        @Nonnull final ProductVariantDraft newProductVariant) {
-        //TODO: Right now it always builds SetPrices UpdateAction, comparison should be calculated GITHUB ISSUE#101.
-        final SetPrices setPricesUpdateAction = SetPrices.of(oldProductVariant.getId(),
-            emptyIfNull(newProductVariant.getPrices()));
-        return singletonList(setPricesUpdateAction);
+        @Nonnull final ProductVariantDraft newProductVariant,
+        @Nonnull final ProductSyncOptions syncOptions) {
+
+        final Map<PriceCompositeId, Price> oldPricesMap =
+            oldProductVariant.getPrices()
+                             .stream()
+                             .collect(Collectors.toMap(PriceCompositeId::of, Function.identity()));
+
+        final List<PriceDraft> newPrices = newProductVariant.getPrices();
+
+        final List<UpdateAction<Product>> updateActions = buildRemoveUpdateActions(oldPricesMap, newPrices,
+            PriceCompositeId::of, price -> RemovePrice.of(price, true));
+
+        final Integer variantId = oldProductVariant.getId();
+
+        emptyIfNull(newPrices).forEach(newPrice -> {
+            if (newPrice == null) {
+                syncOptions.applyErrorCallback(format("Failed to build prices update actions for one price on the "
```

Comment:
How does the user know the product that the variant belongs to? would be this message enough for the user to identify the product and variant?

Reply:
yes :) and the product variant key is unique across a project. (unfortunately, I didn't find this in the API documentation, but by trying it)